### DISCUSSION TITLE
Version argument '--skip-bump-only-releases', avoid bump in fixed mode?

### DISCUSSION THREAD

**Discussion by SchulteDev** [ghiscoding/lerna-lite#928] [Q&A] [answered]
Hi there,

this is about the version command argument '--skip-bump-only-releases' https://www.npmjs.com/package/@lerna-lite/version#--skip-bump-only-releases

> When this option is enabled and a package version is only being bumped without any conventional
> commits detected, the GitHub/GitLab release will be skipped. This will avoid creating releases
> with only "Version bump only for package x" in the release notes, however please note that each
> changelog are still going to be updated with the "version bump only" text.

In my setup there is a monorepo with ‘fixed’ versioning
https://lerna.js.org/docs/features/version-and-publish#fixedlocked-mode-default.
I am using 'skip-bump-only-releases'.
Lerna-lite version creates a new release in this case (updating CHANGELOG files, creating Git tag, pushing changes to Git, ....), even if there is no change according to ConventionalCommits.

```
// lerna.json5
{
  $schema: 'https://raw.githubusercontent.com/lerna-lite/lerna-lite/main/packages/cli/schemas/lerna-schema.json',
  version: '0.3.10',
  loglevel: 'silly',
  packages: [
    'subproject1',
    'subproject2',
    'subproject3',
  ],
  changelogPreset: 'conventional-changelog-conventionalcommits',
  command: {
    version: {
      skipBumpOnlyReleases: true,
      conventionalCommits: true,
      changelogPreset: {
        name: 'conventionalcommits',
      },
      message: 'chore(release): Publish %s\n\nPublishing %s\n[skip ci]',
    },
  },
}

```

```
241a2b203bc73804efb58cda2fe9d4b9b09533e4 chore(release): Publish v0.3.10
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 2631603..6e95e59 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,10 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.3.10](https://dev.azure.com/A-Company/A-Project/_git/some-repository/branchCompare?baseVersion=GTv0.3.9&targetVersion=GTv0.3.10&_a=files) (2024-09-08)
+
+**Note:** Version bump only for package directory-name
+
 ## [0.3.9](https://dev.azure.com/A-Company/A-Project/_git/some-repository/branchCompare?baseVersion=GTv0.3.8&targetVersion=GTv0.3.9&_a=files) (2024-09-08)
 
 ### Bug Fixes 🐛
diff --git a/lerna.json5 b/lerna.json5
index b601165..87a363b 100644
--- a/lerna.json5
+++ b/lerna.json5
@@ -1,6 +1,6 @@
 {
   $schema: 'https://raw.githubusercontent.com/lerna-lite/lerna-lite/main/packages/cli/schemas/lerna-schema.json',
-  version: '0.3.9',
+  version: '0.3.10',
   loglevel: 'silly',
   packages: [
    'subproject1',
diff --git a/subproject2/CHANGELOG.md b/subproject2/CHANGELOG.md
index 4651167..2c5decb 100644
--- a/subproject2/CHANGELOG.md
+++ b/subproject2/CHANGELOG.md
@@ -3,6 +3,10 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.3.10](https://dev.azure.com/A-Company/A-Project/_git/some-repository/branchCompare?baseVersion=GTv0.3.9&targetVersion=GTv0.3.10&_a=files) (2024-09-08)
+
+**Note:** Version bump only for package @A-Company-directory-name/subproject2-root
+
 ## [0.3.9](https://dev.azure.com/A-Company/A-Project/_git/some-repository/branchCompare?baseVersion=GTv0.3.8&targetVersion=GTv0.3.9&_a=files) (2024-09-08)
 
 ### Bug Fixes 🐛
diff --git a/subproject2/package.json b/subproject2/package.json
index 7afbbd4..8023152 100644
--- a/subproject2/package.json
+++ b/subproject2/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@A-Company-directory-name/subproject2-root",
-  "version": "0.3.9",
+  "version": "0.3.10",
   "private": true,
   "files": [
```

I'm not sure if the implementation of the CLI argument is buggy or I'm misunderstanding the CLI flag.

**Workaround**

As a workaround, I am running "lerna version" before doing an actual publish, where "lerna version" is being used for detecting "version bump only" releases. See https://stackoverflow.com/a/78990283/1645517 for details

_Note_: Workaround was updated 2024-09-16

**Comment by ghiscoding** [comment, marked as answer]
You misunderstood the usage of this option, it was only created for independent mode and I just updated the docs to make it clearer. 
https://github.com/lerna-lite/lerna-lite/blob/main/packages/version/README.md#--skip-bump-only-releases

The reason it was created is that in independent mode if you have let say 30 packages and only 10 of them had changes, it was still creating 30 releases but 20 of them had the "bump only.." message and it was flooding the GitHub releases section and so this flag helps with that use case. The flag has no effect at all on fixed mode. Also another thing that was already mentioned in the docs before, even in independent mode with the flag enabled, it will always update the changelogs with the "bump only..." message, what it would however is to simply skip the release in GitHub.

Lerna/Lerna-Lite also has no control on what versions bump is being created when using conventional changelog, because that is all handled by the external library [`conventional-recommended-bump`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-recommended-bump) which will dynamically calculate the next version bump and then pass it back to Lerna.
